### PR TITLE
Make module transformer the last transformer

### PR DESCRIPTION
**Pull request by arv**
This is so that other transformers can output import declarations
in the future.

There are two real changes in here, both related to export
declarations:

ClassTransformer: Make sure that `export default class X {}`
generates valid code.

DestructuringTransformer: Make sure we do not export the temp
variable.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/codegeneration/FromOptionsTransformer.js [hunk 1]
```diff
@@ -238,5 +210,33 @@ export class FromOptionsTransformer extends MultiTransformer {
     if (transformOptions.properTailCalls) {
       append(ProperTailCallTransformer);
     }
+
```
Comment (johnjbarton) [inline src/codegeneration/FromOptionsTransformer.js]: A comment here would remind us of the order-dependence.

File: src/semantics/bindingsInPattern.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ParseTreeVisitor} from '../syntax/ParseTreeVisitor.js';
+import {StringSet} from '../util/StringSet.js';
+
+class BindingsInPattern extends ParseTreeVisitor {
```
Comment (johnjbarton) [inline src/semantics/bindingsInPattern.js]: What is the "InPattern" part  here?

**Comment by johnjbarton** [comment]
LGTM

File: src/semantics/bindingsInPattern.js (hunk 2 above)
Comment (arv) [inline src/semantics/bindingsInPattern.js]: Maybe FindBindingsInPatternVisitor would be better?

**Comment by arv** [comment]
I'll also add a feature test that ensures that a module instance object does not have any extra properties. Before this PR a module instance object would have a `$__0` property if someone exported using destructuring in an export declaration.

File: src/semantics/bindingsInPattern.js (hunk 2 above)
Comment (johnjbarton) [inline src/semantics/bindingsInPattern.js]: FindBindingsInDestructuring?  Neither the class  nor the calllsite talk about patterns so I wasn't able to connect the dots. The callsite does talk about hasDestructuring().  So if it were hasDestructuringPattern(), then BindingsInPattern would be clear.

Comment (arv) [inline src/semantics/bindingsInPattern.js]: Patterns are only used in destructuring at this point. Therefore I didn't see the need to include the word destructuring in there.

File: src/codegeneration/FromOptionsTransformer.js (hunk 1 above)
Comment (arv) [inline src/codegeneration/FromOptionsTransformer.js]: Done.

**Pull request merged**